src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1883,19 +1851,24 @@ private IEnumerable<TypeInferenceInfo> InferTypeInExpressionColon(ExpressionColo
                 return null;
             }
 
-            private IEnumerable<TypeInferenceInfo> InferTypeInNameColon(NameColonSyntax nameColon)
+            private IEnumerable<TypeInferenceInfo> InferTypeInNameColon(NameColonSyntax nameColon, SyntaxToken? previousToken = null)
```

Comment:
no optional parameters.

Reply:
ok, let me change them

Reply:
```
  protected override IEnumerable<TypeInferenceInfo> InferTypesWorker_DoNotCallDirectly(
      SyntaxNode node)
  {
      var expression = node as ExpressionSyntax;
      if (expression != null)
      {
          expression = expression.WalkUpParentheses();
          node = expression;
      }

      var parent = node.Parent;

      return InferTypesWorkerFromParent(parent, node, expression, node.GetFirstToken());
  }
```

Am I allowed to use like in the last line `node.GetFristToken()` as I believe a Node cannot be directly translated to a token. 

Same as in other method I have to get the Node from token which is possible.

Reply:
> Am I allowed to use like in the last line node.GetFristToken() as I believe a Node cannot be directly translated to a token.

No. The apis are distinctly different. The token version is when we're inferring a type for a particular caret *location* (so the token represents what is immediately before the caret.  

The node version is for when we're inferring what the type should be for an invalid node that we're generating a new symbol for.  A toke should not be provided for this case. 

Note: my feedback above was simply that the parameter *not* be *optional*.  It can certainly be nullable.

Reply:
Thanks for the explanation